### PR TITLE
Add AWS Lambda.

### LINKED ISSUE #184: Add Lambda connector
### Description

Add a Lambda connector for FSX. 

### References

_No response_

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by wenjun666**

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -97,6 +106,28 @@ func (p *ONTAPProvider) Schema(ctx context.Context, req provider.SchemaRequest,
 							MarkdownDescription: "Whether to enforce SSL certificate validation, defaults to true",
 							Optional:            true,
 						},
+						"use_aws_lambda_link": schema.BoolAttribute{
+							MarkdownDescription: "Indicates a preference order between several routes to the same destination.",
```
Comment (carchi8py) [inline internal/provider/provider.go]: not sure what this means?

**Review by carchi8py — CHANGES_REQUESTED** [review]

File: internal/restclient/rest_response.go [hunk 2]
```diff
@@ -27,6 +27,121 @@ type RestResponse struct {
...
+		emptyResponse.ErrorType = "bad_response_decode_json"
+		return statusCode, emptyResponse, err
+	}
+	tflog.Debug(c.ctx, fmt.Sprintf("awsDataMap %#v", awsDataMap))
+	statusCode = int(awsDataMap["status"].(float64))
+
+	// type awsStagedResponse struct {
+	// 	Status int               `mapstructure:"status"`
+	// 	Data   AWSLambdaRestData `mapstructure:"data"`
+	// }
```
Comment (suhasbshekar) [inline internal/restclient/rest_response.go]: can you remove the commented code ?

**Review by suhasbshekar — CHANGES_REQUESTED** [review]

**Review by carchi8py — APPROVED** [review]

File: internal/provider/provider.go [hunk 3]
```diff
@@ -97,6 +109,30 @@ func (p *ONTAPProvider) Schema(ctx context.Context, req provider.SchemaRequest,
 							MarkdownDescription: "Whether to enforce SSL certificate validation, defaults to true",
 							Optional:            true,
 						},
+						"aws": schema.SingleNestedAttribute{
+							MarkdownDescription: "AWS configuration for Lambda",
+							Optional:            true,
+							Attributes: map[string]schema.Attribute{
+								"lambda": schema.SingleNestedAttribute{
```
Comment (carchi8py) [inline internal/provider/provider.go]: Change to Just Aws

**Review by carchi8py — CHANGES_REQUESTED** [review]

**Comment by chuyich** [comment]
The lint issue can be resolved after the PR sync with the latest integration/main branch.

**Comment by chuyich** [comment]
Missing CHANGELOG

**Review by chuyich — CHANGES_REQUESTED** [review]
Missing CHANGELOG and documentation.

**Review by carchi8py — COMMENTED** [review]
Looks good can you make the change to remove commented out code that @suhasbshekar mentioned.

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ FEATURES:
 ENHANCEMENTS:
 * **netapp-ontap_lun**: added `size_unit` option. ([#227](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/227))
 * **netapp-ontap_security_account**: Add support for import and update ([#243](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/243))
+* **provider**: add `aws_lambda` option. ([#262](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/262))
```
Comment (carchi8py) [inline CHANGELOG.md]: Can you put this under features and at the top

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,7 +13,7 @@ FEATURES:
 
 ENHANCEMENTS:
 * **netapp-ontap_lun**: added `size_unit` option. ([#227](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/227))
-* **netapp-ontap_security_account**: Add support for import and update ([#243](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/243))
+* **netapp-ontap_security_account**: Add support for import and update ([#243](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/243)
```
Comment (carchi8py) [inline CHANGELOG.md]: missing a 
)
at the end of the line

**Review by carchi8py — CHANGES_REQUESTED** [review]

**Review by suhasbshekar — APPROVED** [review]

**Review by carchi8py — APPROVED** [review]

**Review by carchi8py — CHANGES_REQUESTED** [review]
You need to update doc/index.html with the documentation for the new variables. 

So this page https://github.com/NetApp/terraform-provider-netapp-ontap/blob/integration/main/docs/index.md show that new information.

**Review by carchi8py — APPROVED** [review]

**Review by chuyich — APPROVED** [review]

**Pull request merged**